### PR TITLE
[2.0.0] Auto-add source id identifiers to all robot works

### PR DESCRIPTION
**Pull request by aaxelb**

### CODE REVIEW

File: share/normalize/normalizer.py [hunk 1]
```diff
@@ -71,13 +72,39 @@ def get_root_parser(self):
...
+        if source_id:
+            self.add_source_identifier(source_id, jsonld, root_ref)
+
         ctx.clear()  # Clean up
 
         return jsonld
+
+    def add_source_identifier(self, source_id, jsonld, root_ref):
+        uri = IRILink(urn_fallback=True).execute(str(source_id))['IRI']
+        if any(n for n in jsonld['@graph'] if n['@type'] == 'workidentifier' and n['uri'] == uri):
```
Comment (chrisseto) [inline share/normalize/normalizer.py]: May want to add a `.lower()` on `n['@type']`

File: share/normalize/normalizer.py [hunk 2]
```diff
@@ -71,13 +72,39 @@ def get_root_parser(self):
...
+            return
+
+        root_node = next(n for n in jsonld['@graph'] if root_ref.items() <= n.items())
```
Comment (chrisseto) [inline share/normalize/normalizer.py]: `ctx.pool[root_ref]`

File: share/normalize/normalizer.py [hunk 3]
```diff
@@ -71,13 +72,39 @@ def get_root_parser(self):
...
+        identifier_ref = {'@id': '_:' + uuid.uuid4().hex, '@type': 'workidentifier'}
+        identifier = {
+            'uri': uri,
+            'creative_work': root_ref,
+            **identifier_ref
+        }
+
+        try:
+            root_node['identifiers'].append(identifier_ref)
+        except KeyError:
```
Comment (chrisseto) [inline share/normalize/normalizer.py]: `root_node['...'].setdefault(key, []).append(...)`

File: share/normalize/normalizer.py (hunk 2 above)
Comment (aaxelb) [inline share/normalize/normalizer.py]: Oh nice, forgot that was still available here.

File: share/normalize/normalizer.py (hunk 3 above)
Comment (aaxelb) [inline share/normalize/normalizer.py]: 👍

**Comment by aaxelb** [comment]
@chrisseto, your concerns are addressed

**Pull request merged**